rfcs/text/0430-hook-service.md

```diff
@@ -0,0 +1,148 @@
+
+When receiving `null` for the `node`, the embedding application should tear down any and all subscriptions made in association with this hook.
+
+As this service relies on the embedding application directly manipulating DOM nodes, it will not be possible to implement this where block's 'proxy' the Block Protocol using `postMessage` (i.e, where a block includes an `iframe`). This service requires that the embedding application implementation of the service share the global environment (commonly referred to as a 'scope' or 'realm') with the DOM node passed by the block in the `hook` message.
+
+### Errors
+
+Where an embedding application has not implemented a hook, it must respond with a `NOT_IMPLEMENTED` error, which will allow blocks to implement a fallback. This error must include the `node`, `type`, `path` and `hookId` properties passed in the original `hook` message.
+
+### Notes
+
+- This RFC makes no changes to the Graph Service's requirements that an embedding application implementing the Graph Service must provide data in the format specified by a block's schema.
+- The embedding application remains in control of data, as specified in the Graph Service, and should continue to indicate whether the block is in a readonly state as specified in that service. The implementation of the Hook Service should mirror this state.
+
+## Core Specification changes
+
+When an embedding application receives a message from a block where it does the implement the `service` specified in the message, it must respond with a `NOT_IMPLEMENTED` error, so that blocks can respond appropriately.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+TODO:
+
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+- Its interaction with other features is clear.
+- It is reasonably clear how the feature would be implemented.
+- Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- There is the possibility for explosion in values used in `type`, or for blocks to use it to tie their block to a specific embedding application, or set of embedding applications.
+- The core specification which services build on does not allow sync communication – which may be an issue when it comes to rendering UI. This RFC does not propose changing this at this time as we can investigate in the implementation of the Hook Service if this becomes a problem.
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+TODO:
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+
+# Prior art
+
+[prior-art]: #prior-art
```

Comment:
Suggest making the discussion of `editableRef` the 'rationale and alternatives' as it includes both an alternative (`editableRef`) and the rationale for moving away from it.
```suggestion
```